cypress/integration/step_definitions/Ui/single_vanities.cy.js

```diff
@@ -16,6 +14,51 @@ When('User clicks to Single Vanities', () => {
 });
    
 
+// Scenario: Color filter test 
+
+let singleVanitiesPage=null
+before(()=>{
+    singleVanitiesPage =new SingleVanitiesPage();
+})
+
+
+  And ('User clicks the Color Filter Button', () => {
+    singleVanitiesPage.clickColorButton();
+  });
+  
+  And ('User clicks Green colour', () => {
+    singleVanitiesPage.clickGreenColour();
+  })
+
+  Then ('The page should display only the vanities which are green' , () => {
```

Comment:
Instead of picking green, would it be possible to make this more dynamic? Imagine they added a new color and instead of writing new function, this function should work with that new color just adding this color as parameter from feature file.

Reply:
yes it would be great :)) I will try